addons/website_event_sale/controllers/main.py

```diff
@@ -62,6 +62,12 @@ def registration_confirm(self, event, **post):
         # we have at least one registration linked to a ticket -> sale mode activate
         if any(info['event_ticket_id'] for info in registrations):
             order_sudo = request.website.sale_get_order()
+            if order_sudo.partner_id.is_public:
+                contact = tools.formataddr((registrations[0]['name'], registrations[0]['email']))
+                partner = request.env['res.partner'].sudo().find_or_create(contact)
+                if not partner.phone and registrations[0].get('phone'):
+                    partner.phone = registrations[0]['phone']
+                order_sudo.partner_id = partner
```

Comment:
But doesn't that mean that the partner_id of the registration will not match the partner set on the SO?

Either we decide to start creating partners every time someone registers and set it on the registration record and then we use that partner to sync with the SO (which I don't think we want), either we keep it as is now, but this seems wrong.